src/main/java/com/example/App.java

```diff
@@ -0,0 +1,38 @@
+package com.example;
+
+import lombok.SneakyThrows;
+
+import javax.sql.DataSource;
+
+public class App {
+    @SneakyThrows
+    public static void main(String[] args) {
+        var dataSource = initializePooledDataSource();
+        var total = 0.0;
+        var start = System.nanoTime();
+        for (int i = 0; i < 500; i++) {
+            try (var connection = dataSource.getConnection()) {
+                connection.setAutoCommit(false);
+                try (var statement = connection.createStatement()) {
+                    var rs = statement.executeQuery("select random() from products");
+                    rs.next();
+                    total += rs.getDouble(1);
+                }
+                connection.rollback();
+            }
+        }
+        System.out.println((System.nanoTime() - start) / 1000_000 + " ms");
+        System.out.println(total);
+
+
+    }
+
+    private static DataSource initializePooledDataSource() {
+        int threadCount = 10;
```

Comment:
It's a connection count, not a thread count. 😉

Reply:
Also, not datasourcePool but connectionPool))) Seems like I absolutely didn't care of naming, haha